File: 203_remove_linked_list_elements/removeElements.py
class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        tmp = head
        prev = None
        
        while tmp:
            if tmp.val == val:
                if prev == None:
                    head = head.next
                else:
                    prev.next = tmp.next
            else:
                prev = tmp
            
            tmp = tmp.next


        return head

File: 203_remove_linked_list_elements/removeElements.cpp
#include <iostream>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x): val(x), next(NULL) {}
};

class Solution {
	public:
    	ListNode* removeElements(ListNode* head, int val) {
            ListNode *dummy = new ListNode(val + 1) ;
            dummy -> next = head;
            ListNode *temp = dummy, *prev = NULL;
            while(temp) {
                if(temp -> val == val) {
                    prev -> next = temp -> next;
                }
                else prev = temp;
                temp = temp -> next;
            }

            return dummy -> next;
    	}
};
